File: examples/java-maven/src/main/java/com/example/models/Status.java
package com.example.models;

/**
 * Status enumeration for entities.
 */
public enum Status {
    ACTIVE("active"),
    INACTIVE("inactive"),
    PENDING("pending");

    private final String value;

    Status(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static Status fromValue(String value) {
        for (Status status : values()) {
            if (status.value.equals(value)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown status: " + value);
    }

    public boolean isActive() {
        return this == ACTIVE;
    }

    public boolean isInactive() {
        return this == INACTIVE;
    }

    public boolean isPending() {
        return this == PENDING;
    }
}

File: examples/kotlin-gradle/src/main/kotlin/com/example/models/Status.kt
package com.example.models

import kotlinx.serialization.Serializable

/**
 * Status enumeration for entities.
 */
@Serializable
enum class Status(val value: String) {
    ACTIVE("active"),
    INACTIVE("inactive"),
    PENDING("pending");

    val isActive: Boolean get() = this == ACTIVE
    val isInactive: Boolean get() = this == INACTIVE
    val isPending: Boolean get() = this == PENDING

    companion object {
        fun fromValue(value: String): Status =
            entries.find { it.value == value }
                ?: throw IllegalArgumentException("Unknown status: $value")
    }
}
